lib/handler/configurator/proxy.c

```diff
@@ -142,6 +142,52 @@ static int on_config_ssl_cafile(h2o_configurator_command_t *cmd, h2o_configurato
     return ret;
 }
 
+static int on_config_ssl_session_cache(h2o_configurator_command_t *cmd, h2o_configurator_context_t *ctx, yoml_t *node)
+{
+    struct proxy_configurator_t *self = (void *)cmd->configurator;
+    size_t i;
+
+    switch (node->type) {
+    case YOML_TYPE_SCALAR:
+        if (strcasecmp(node->data.scalar, "OFF") == 0) {
+            /* disabled */
+            self->vars->session_cache.capacity = 0;
+        } else if (strcasecmp(node->data.scalar, "ON") == 0) {
+            /* use default values */
+        } else {
+            h2o_configurator_errprintf(cmd, node, "scalar argument must be either of: `OFF`, `ON`");
+            return -1;
+        }
+        break;
+    case YOML_TYPE_MAPPING:
+        for (i = 0; i != node->data.mapping.size; ++i) {
+            yoml_t *key = node->data.mapping.elements[i].key;
+            yoml_t *value = node->data.mapping.elements[i].value;
+            if (key->type == YOML_TYPE_SCALAR) {
+                if (strcasecmp(key->data.scalar, "capacity") == 0) {
+                    if (h2o_configurator_scanf(cmd, value, "%" PRIu64, &self->vars->session_cache.capacity) != 0)
+                        return -1;
+                } else if (strcasecmp(key->data.scalar, "lifetime") == 0) {
+                    if (h2o_configurator_scanf(cmd, value, "%" PRIu32, &self->vars->session_cache.lifetime) != 0)
+                        return -1;
+                } else {
+                    goto InvalidKey;
```

Comment:
Please please do not abuse goto. Goto should never be used for jumping into another block.